src/store/__test__/game.test.ts

```diff
@@ -0,0 +1,253 @@
+      expect(ScryfallApiInstance.getRandomArt).not.toBeCalled();
+    });
+
+    it("fetches random card and random art from that card", async () => {
+      const game = useGameStore();
+      game.query = {
+        search: "Search Query",
+      };
+
+      const res = await game.fetchNextCard();
+
+      expect(res).toEqual(CardBank.Jolene);
+      expect(ScryfallApiInstance.getRandomCard).toBeCalledTimes(1);
+      expect(ScryfallApiInstance.getRandomCard).toBeCalledWith("Search Query", {
+        includeExtras: undefined,
+      });
+      expect(ScryfallApiInstance.getRandomArt).toBeCalledTimes(1);
+      expect(ScryfallApiInstance.getRandomArt).toBeCalledWith(
+        CardBank.Alesha.oracle_id,
+        "Search Query"
+      );
+    });
+
+    it("uses previous oracle id if passed", async () => {
+      const game = useGameStore();
+      game.query = {
+        search: "Search Query",
+      };
+
+      await game.fetchNextCard("abc-123");
+
+      expect(ScryfallApiInstance.getRandomCard).toBeCalledTimes(1);
+      expect(ScryfallApiInstance.getRandomCard).toBeCalledWith("Search Query -oracle_id:abc-123", {
+        includeExtras: undefined,
+      });
+      expect(ScryfallApiInstance.getRandomArt).toBeCalledTimes(1);
+      expect(ScryfallApiInstance.getRandomArt).toBeCalledWith(
+        CardBank.Alesha.oracle_id,
+        "Search Query -oracle_id:abc-123"
+      );
+    });
+
+    it("passes along includeExtras value to getRandomCard", async () => {
+      const game = useGameStore();
+      game.query = {
+        search: "Search Query",
+        includeExtras: true,
+      };
+
+      await game.fetchNextCard("abc-123");
+
+      expect(ScryfallApiInstance.getRandomCard).toBeCalledTimes(1);
+      expect(ScryfallApiInstance.getRandomCard).toBeCalledWith("Search Query -oracle_id:abc-123", {
+        includeExtras: true,
+      });
+      expect(ScryfallApiInstance.getRandomArt).toBeCalledTimes(1);
+      expect(ScryfallApiInstance.getRandomArt).toBeCalledWith(
+        CardBank.Alesha.oracle_id,
+        "Search Query -oracle_id:abc-123"
+      );
+    });
+
+    it("manages card state for successful queries", async () => {
+      const game = useGameStore();
+
+      vi.mocked(ScryfallApiInstance.getRandomArt).mockImplementation(() => {
+        return new Promise((resolve) => {
+          finishApiCall = resolve;
+        });
+      });
+      game.query = { search: "query" };
+      game.card = CardBank.ArborElf;
+
+      const fetchPromise = game.fetchNextCard();
+      await nextTick();
+
+      expect(game.nextCardStatus).toEqual(LoadingStatus.Pending);
+      expect(game.previousCard).toEqual(CardBank.ArborElf);
+      expect(game.card).toEqual(CardBank.ArborElf);
+
+      finishApiCall(CardBank.Jolene);
+      await fetchPromise;
+
+      expect(game.nextCardStatus).toEqual(LoadingStatus.Success);
+      expect(game.previousCard).toEqual(CardBank.ArborElf);
+      expect(game.card).toEqual(CardBank.Jolene);
+    });
```

Comment:
Don't love how this pattern of delaying the api response this way, but it was the best way I could figure out to assert on the state of the store before the api call finished. Open to suggestions though.